### PR TITLE
Feature/revsdl 2383 gpb data

### PR DESCRIPTION
**Pull request by bonewell**

### CODE REVIEW

File: src/components/vr_module/include/vr_module/vr_proxy.h [hunk 1]
```diff
@@ -63,9 +63,16 @@ class VRProxy : public threads::MessageLoopThread<MessageQueue>::Handler {
...
  private:
+  void Receive();
   void Handle(vr_hmi_api::Message message);
+  void OnReceived(const vr_hmi_api::Message& message);
+  std::string SizeToString(int32_t value);
+  int32_t SizeFromString(const std::string& value);
   VRProxyListener *listener_;
   threads::MessageLoopThread<MessageQueue> incoming_;
+  Channel *channel_;
+  threads::Thread* channel_thread_;
```
Comment (KKarlash) [inline src/components/vr_module/include/vr_module/vr_proxy.h]: +  Channel *channel_;
 +  threads::Thread* channel_thread_;

Please keep common style: * everywhere near with name or everywhere near with type :)

File: src/components/vr_module/net/src/connected_socket_impl.cc [hunk 2]
```diff
@@ -0,0 +1,218 @@
+#include "net/connected_socket_impl.h"
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/un.h>
+#include <arpa/inet.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
```
Comment (KKarlash) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: section with includes should be sorted

File: src/components/vr_module/net/src/connected_socket_impl.cc [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+  /* The CMSG macros yused to create and access control
+     messages (which is another name for ancillary data). The ancillary
+     data is made up of pairs of struct cmsghdr structures and associated
+     data arrays. First, we're going to allocate a buffer on the stack to
+     contain our
+     data array (that contains the socket).
+  */
+  struct cmsghdr* cmp;
+  cmp = static_cast<cmsghdr*>(malloc(CMSG_SPACE(sizeof(int))));
```
Comment (KKarlash) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: could you please rewrite in c++?

File: src/components/vr_module/net/src/connected_socket_impl.cc [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+     put our file descriptor in.
+  */
+  Int32* control_msg =
+  reinterpret_cast<Int32*>(CMSG_DATA(control_message));
+  *control_msg = file_fd;
+
+  // send the file descriptor
+  if (-1 == (ret = ::sendmsg(socket_handle(), &message, flags))) {
+    LOG4CXX_ERROR(logger_, "Unable to send " << strerror(errno));
+    free(cmp);
```
Comment (KKarlash) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: could you please rewrite in c++?

File: src/components/vr_module/net/src/connected_socket_impl.cc [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+  }
+
+  free(cmp);
```
Comment (KKarlash) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: could you please rewrite in c++?

File: src/components/vr_module/net/src/connected_socket_impl.cc [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+  /* We also have to update the controllen in case other stuff is actually
+     in there we may not be aware of (due to macros)
+  */
+  message.msg_controllen = control_message->cmsg_len;
+
+  /* get a pointer to the start of the ancillary data array and
+     put our file descriptor in.
+  */
+  Int32* control_msg =
+  reinterpret_cast<Int32*>(CMSG_DATA(control_message));
```
Comment (KKarlash) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: is it necessary?

File: src/components/vr_module/src/socket_channel.cc [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  if (socket_) {
+    socket_->close();
+  }
+  delete socket_;
+}
+
+bool SocketChannel::Start() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::string address = "127.0.0.1";
+  UInt32 port = 5431;
```
Comment (KKarlash) [inline src/components/vr_module/src/socket_channel.cc]: Maybe it's better to use constants for these strings?

**Comment by KKarlash** [comment]
Reviewed

File: src/components/vr_module/net/src/connected_socket_impl.cc (hunk 3 above)
Comment (DKlimenko) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: Kosta it is CM code, we just decided to integrate it to RSDL

File: src/components/vr_module/net/src/connected_socket_impl.cc (hunk 4 above)
Comment (DKlimenko) [inline src/components/vr_module/net/src/connected_socket_impl.cc]: Kosta it is CM code, we just decided to integrate it to RSDL

**Comment by VladSemenyuk** [comment]
Reviewed

**Pull request closed without merging**